compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/InnerClassesLowering.kt

```diff
@@ -91,6 +95,16 @@ class InnerClassesLowering(val context: BackendContext) : ClassLoweringPass {
                     enclosingConstructor = null
                 }
 
+            override fun visitFunction(declaration: IrFunction): IrStatement =
+                try {
```

Comment:
This way of handling context looks ad-hoc and brittle to me. Why not use `IrElementTransformerVoidWithContext`?

Reply:
Good question! I changed it to use `IrElementTransformerVoidWithContext`. This doesn't change the fact that we don't deal with nested functions, but it does make the code more compact.